src/main/java/com/vg/live/SuperLive.java

```diff
@@ -181,7 +181,7 @@ public static boolean isEOF(ByteBuffer buf) {
         int sampleFlags = tfhd.getDefaultSampleFlags();
 
         for (int i = 0; i < sampleCount; i++) {
-            if (!trun.isFirstSampleFlagsAvailable()) {
+            if (trun.getSamplesFlags() != null) {
```

Comment:
what if sample sizes are explicitly set in trun? would you read them if trun.getSamplesFlags() == null ?

Reply:
@chugai sizes, flags, and duration are in the same table.
And I am assuming that If samplesFlags[] == null then sampleSizes and sampleDurations arrays are null

Reply:
> And I am assuming that If samplesFlags[] == null then sampleSizes and sampleDurations arrays are null
not a valid assumption

Reply:
and first sample flag support?